sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/LongOffset.scala

```diff
@@ -22,8 +22,18 @@ package org.apache.spark.sql.execution.streaming
  */
 case class LongOffset(offset: Long) extends Offset {
 
+  override val json = offset.toString
+
   def +(increment: Long): LongOffset = new LongOffset(offset + increment)
   def -(decrement: Long): LongOffset = new LongOffset(offset - decrement)
 
   override def toString: String = s"#$offset"
 }
+
+object LongOffset {
+
+  def apply(serialized: Offset) : LongOffset = new LongOffset(serialized.json.toLong)
+
+}
```

Comment:
Can you please clarify best practices w.r.t. extra lines? I didn't see anything in the code style guide.